importer/api/__init__.py

```diff
@@ -124,6 +125,7 @@ def import_course(bundle, repo_id, user_id, static_dir):
     )
     import_static_assets(course, static_dir)
     import_children(course, src, None)
+    populate_xanalytics_fields.delay(course.id)
```

Comment:
There is no layering of async on async; there's a single queue. Throwing a task onto the queue from a worker doesn't involve any nesting or recursion. Given that, I don't see any functional difference between the current version and the suggested version, and the current version is simpler (less code).

I think it's a bad idea to list out all "expected" exception types for the reason I stated above. A custom exception would work, but would require a "bare Except" to ensure everything was caught.

I think the main difference is that you think there's something inherently wrong about scheduling a Celery task from within a Celery task, and I don't. Unless you have a reason it's bad that I'm not seeing, I'd prefer to leave this as it is.